vnext/Shared/Networking/WinRTHttpResource.cpp

```diff
@@ -494,37 +518,62 @@ WinRTHttpResource::PerformSendRequest(HttpMethod &&method, Uri &&rtUri, IInspect
             self->m_onRequestSuccess(reqArgs->RequestId);
           }
 
+          if (self->m_onComplete) {
+            self->m_onComplete(reqArgs->RequestId);
+          }
           co_return;
         }
       }
 
-      auto isText = reqArgs->ResponseType == "text";
       if (isText) {
         reader.UnicodeEncoding(UnicodeEncoding::Utf8);
       }
 
-      // #9510
+      int64_t received = 0;
       string responseData;
       winrt::Windows::Storage::Streams::IBuffer buffer;
       while (auto loaded = co_await reader.LoadAsync(segmentSize)) {
         auto length = reader.UnconsumedBufferLength();
+        received += length;
 
         if (isText) {
           auto data = vector<uint8_t>(length);
           reader.ReadBytes(data);
 
-          responseData += string(Common::Utilities::CheckedReinterpretCast<char *>(data.data()), data.size());
+          auto increment = string(Common::Utilities::CheckedReinterpretCast<char *>(data.data()), data.size());
+          // #9534 - Send incremental updates.
+          // See https://github.com/facebook/react-native/blob/v0.70.6/Libraries/Network/RCTNetworking.mm#L561
+          if (reqArgs->IncrementalUpdates) {
+            responseData = std::move(increment);
+
+            if (self->m_onIncData) {
```

Comment:
>self->m_onIncData [](http://example.com/codeflow?start=16&length=17)

Should we fail much earlier before making networking request if IncrementalUpdates flag is on but these callbacks are not registered ?

Reply:
Not necessarily.
The requests should still get through even if they don't get registered to the JS layer (i.e. unit test scenarios).

For production scenarios, the callbacks are always set (see HttpModule.cpp).